editor/src/clj/editor/lsp.clj

```diff
@@ -868,22 +867,38 @@
     ;; Don't ask the servers about dependency (zip) resources
     (do (result-callback {:complete true :items []}) nil)))
 
+(defn capability-resolve-completions? [capability]
+  (-> capability :completion :resolve boolean))
+
 (defn resolve-completion! [lsp completion result-callback & {:keys [timeout-ms]
                                                              :or {timeout-ms 5000}}]
   (if-let [completion-out (:editor.lsp.server-state/out (meta completion))]
     (lsp (bound-fn [state]
            (let [ch (a/chan 1)]
              (a/go (result-callback (or (<! ch) completion)))
-             (send-requests! state
-                             ch
-                             :requests-fn (fn [_ {:keys [out]}]
-                                            (when (= out completion-out)
-                                              [(lsp.server/resolve-completion
-                                                 completion
-                                                 #(vary-meta % dissoc :editor.lsp.server-state/out))]))
-                             :timeout-ms timeout-ms))))
+             (send-requests!
+               state ch
+               :capabilities-pred capability-resolve-completions?
```

Comment:
This is not related to hovers, but this is something I noticed when I was trying to add the glsl analyzer as a built-in language server: servers announce if they support resolving completion items, and we should take it into account.